pages/components/button.html

```diff
@@ -0,0 +1,151 @@
+<html lang="en">
+  <%- include('head.html'); -%>
+  <body>
+    <f-docs-template>
+      <%- include('nav.html'); -%>
+      <main slot="content">
+        <h1 class="mb-16">Button</h1>
+        <p>Buttons are used to perform actions, with different visuals for different needs.</p>
+
+        <h2 class="mt-24 mb-16">Props</h2>
+        <table class="w-full docs-table">
+          <thead>
+            <tr class="text-left text-gray-400 uppercase">
+              <th>prop</th>
+              <th>type</th>
+              <th>default</th>
+            </tr>
+          </thead>
+          <tbody class="align-top">
+            <tr>
+              <td>type</td>
+              <td>
+                <div>"button" | "submit" | "reset"</div>
+                <div>Button type, only applied when href is not set.</div>
+              </td>
+              <td>"button"</td>
+            </tr>
+            <tr>
+              <td>variant</td>
+              <td>"primary" | "secondary" | "negative" | "utility" | "link" | "pill"</td>
+              <td>"secondary"</td>
+            </tr>
+            <tr>
+              <td>quiet</td>
+              <td>boolean</td>
+              <td>false</td>
+            </tr>
+            <tr>
+              <td>small</td>
+              <td>boolean</td>
+              <td>false</td>
+            </tr>
+            <tr>
+              <td>loading</td>
+              <td>boolean</td>
+              <td>false</td>
+            </tr>
+            <tr>
+              <td>href</td>
+              <td>
+                <div>string</div>
+                <div>Set the href for the location where clicking the button will take you to.</div>
+              </td>
+              <td>-</td>
+            </tr>
+            <tr>
+              <td>target</td>
```

Comment:
Is there a reason why `target` and `rel` are props? Does Lit not have a way of forwarding extra attributes to a target element?

Asking because if not, this means some extra attribute like `referrerpolicy` would just not work?

Reply:
> Note: rel I kinda get because we want to set a sane default when href is truthy, but target makes me question things then 

I've just found out that since 2021 browsers implicitly set `rel="noopener"` for any `target="_blank"` links:
https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/noopener

I'm still investigating how to approach the support for extra attributes. Looking at [Shoelace](https://shoelace.style/components/button), [Fast](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/button/button.spec.md) and [Ionic](https://ionicframework.com/docs/api/button) and they seem to solve this in different ways, mostly accepting a limited number of attributes. I'm considering leaning towards what Fast does, i.e. providing an anchor and button component separately, accepting all respective attributes. That would at least respect the rule of [One Web Component, one responsibility](https://medium.com/ing-blog/best-practices-to-build-web-components-6d517923fba4).